### DISCUSSION TITLE
Replication strategy between S3 and MinIO

### DISCUSSION THREAD

**Discussion by vicmunoz** [minio/minio#18502] [Q&A] [answered]
Hi everyone, 

I'm looking for advice on strategy on how to replicate AWS S3 bucket as source to on-premise minio deployment. 

The numbers are 6TB total, around 40M objects, with tree structure of 150k folders on first level and no more 5 folders of depth. 
We got around 25k new objects per day. 

I already replicated all data with minio batch jobs based on prefix and filters for newly added objects. However when thinking how to maintain replica updated, I don't see minio mirror --watch enough reliable. All my attempts to run mirror ended with job running for more than an hour without replicating nothing. 

Could you explain a little bit what's under the carpet of mirror process? How is scanning for newly added objects?
For dataset with high numbers like above, maybe a strategy of S3 events is more reliable? I mean, for every object update in source send event and target acting accordingly.

Thanks in advance

**Comment by harshavardhana** [comment, marked as answer]
https://min.io/docs/minio/linux/administration/batch-framework.html
